src/json_writer.rs

```diff
@@ -84,33 +95,51 @@ impl ZipArchiveJsonWriter {
     }
 }
 
+// We are going to use Option types here to denote that not
+// all of these are going to be calculated, based on certain
+// command-line flags given to this program.
+//
+// TODO: Figure out how to get the JSON serializer to dump fields
```

Comment:
Are you satisfying the `TODO` with the `#[serde(skip_serializing_if="Option::is_none")]`?